DuckDuckGo/SettingsViewModel.swift

```diff
@@ -217,40 +212,64 @@ extension SettingsViewModel {
-                hasActiveSubscription = false
-#endif
-                return SettingsState.PrivacyPro(enabled: enabled,
-                                                canPurchaseSubscription: canPurchaseSubscription,
-                                                hasActiveSubscription: hasActiveSubscription)
-            }()
+            networkProtection: getNetworkProtectionState(),
+            privacyPro: getPrivacyProState(),
+            sync: getSyncState()
         )
+        
         setupSubscribers()
-#if SUBSCRIPTION
+        
+        #if SUBSCRIPTION
         if #available(iOS 15, *) {
-            Task { await setupSubscriptionEnvironment() }
+            Task {
+                if state.privacyPro.enabled {
+                    await setupSubscriptionEnvironment()
+                }
+            }
         }
-#endif
+        #endif
+    }
+    
+    private func getNetworkProtectionState() -> SettingsState.NetworkProtection {
+        var enabled = false
+        #if NETWORK_PROTECTION
+            if #available(iOS 15, *) {
+                let accessController = NetworkProtectionAccessController()
+                enabled = accessController.networkProtectionAccessType() != .none
+            }
+        #endif
+        return SettingsState.NetworkProtection(enabled: enabled, status: "")
+    }
+    
+    private func getPrivacyProState() -> SettingsState.PrivacyPro {
+        var enabled = false
+        var canPurchase = false
+        var hasActiveSubscription = Self.cachedHasActiveSubscription
+        #if SUBSCRIPTION
+            enabled = featureFlagger.isFeatureOn(.privacyPro)
+            canPurchase = SubscriptionPurchaseEnvironment.canPurchase
+        #endif
+        return SettingsState.PrivacyPro(enabled: enabled,
+                                        canPurchase: canPurchase,
+                                        hasActiveSubscription: hasActiveSubscription)
+    }
+    
+    private func getSyncState() -> SettingsState.SyncSettings {
+        SettingsState.SyncSettings(enabled: legacyViewProvider.syncService.featureFlags.contains(.userInterface),
+                                 title: {
+                                     let syncService = legacyViewProvider.syncService
+                                     let isDataSyncingDisabled = !syncService.featureFlags.contains(.dataSyncing)
+                                      && syncService.authState == .active
+                                     if SyncBookmarksAdapter.isSyncBookmarksPaused
+                                         || SyncCredentialsAdapter.isSyncCredentialsPaused
+                                         || isDataSyncingDisabled {
+                                         return "⚠️ \(UserText.settingsSync)"
+                                     }
+                                     return UserText.settingsSync
+                                 }())
```

Comment:
👍

Reply:
Please just rename `UserText.settingsSync` to `SyncUI.UserText.syncTitle` (importing SyncUI in this file as needed). UserText.settingsSync should be removed. Thanks!